File: ViewBase/Source/Validators/EmptyStringValidator.vb
Imports FluentValidation

Public Class EmptyStringValidator
    Inherits AbstractValidator(Of String)
    Sub New(propertyDisplayName As String)
        RuleFor(Function(text) text).
            Must(Function(text) Not String.IsNullOrEmpty(text)).WithMessage($"{propertyDisplayName}: no puede quedar vacío.")
    End Sub
End Class

File: BaseView/src/Validators/EmptyStringValidator.cs
using FluentValidation;

namespace Wells.BaseView.Validators
{
    public class EmptyStringValidator : AbstractValidator<string>
    {
        public EmptyStringValidator(string propertyDisplayName)
        {
            RuleFor((text) => text).Must((text) => !string.IsNullOrEmpty(text)).WithMessage($"{propertyDisplayName}: no puede quedar vacío.");
        }
    }
}
